src/basic/chase.c

```diff
@@ -546,7 +546,7 @@ int chase(const char *path, const char *original_root, ChaseFlags flags, char **
         return r;
 }
 
-int chase_and_open(const char *path, const char *root, ChaseFlags chase_flags, int open_flags, char **ret_path) {
+int chase_and_open(const char *path, const char *root, ChaseFlags chase_flags, int64_t open_flags, char **ret_path) {
```

Comment:
urks. also ugly. i would not try to shoehorn this locking stuff into the usual O_xyz flags.

just add a new flags param to xopenat() and then add in chase() and friends add matching flags to the existing ChaseFlags and convert them from ChaseFlags to xopenat()s new flag param as needed.